golang/internal/helper/image/image_unit_test.go

```diff
@@ -13,35 +13,50 @@ import (
 	"github.com/dyrector-io/dyrectorio/protobuf/go/agent"
 )
 
-func TestRegistryUrl(t *testing.T) {
-	auth := &imageHelper.RegistryAuth{
-		URL: "test",
-	}
-
-	url := imageHelper.GetRegistryURL(nil, auth)
-	assert.Equal(t, url, "test")
+type RegistryTestCase struct {
+	Registry    *string
+	RegistryUrl *string
+	ExpectedUrl string
 }
 
-func TestRegistryUrlPriority(t *testing.T) {
-	registry := "other"
-	auth := &imageHelper.RegistryAuth{
-		URL: "test",
-	}
-
-	url := imageHelper.GetRegistryURL(&registry, auth)
-	assert.Equal(t, url, "test")
+func NewPTR[T any](value T) *T {
```

Comment:
This is a nice general helper function, this should be moved higher up in the hierarchy.
`golang/internal/pointer/pointer.go` it would be nice, with a simple unit test in addition.

Reply:
Glad you like it, i was doubtful about it. I extracted the code to `golang/internal/pointer/pointer.go`